src/motion-queue-motor-operations_test.cc

```diff
@@ -202,6 +205,32 @@ TEST(RealtimePosition, zero_loops_edge) {
   EXPECT_THAT(expected, ::testing::ContainerEq(status.pos_steps));
 }
 
+// Clear motion queue motor operations.
+// The physical status should be reset and motion_backend.HaltAndDiscard()
+// called.
+TEST(RealtimePosition, clear_queue) {
+  HardwareMapping hw;
+  MockMotionQueue motion_backend = MockMotionQueue();
+  MotionQueueMotorOperations motor_operations(&hw, &motion_backend);
+
+  // Enqueue a segment
+  LinearSegmentSteps segment =
+    CreateMockSegment({10, 20, 30, 40, 50, 60, 70, 80});
+  int expected[BEAGLEG_NUM_MOTORS] = {10, 20, 30, 40, 50, 60, 70, 80};
+
+  motor_operations.Enqueue(segment);
+  motion_backend.SimRun(0, 1);
+
+  PhysicalStatus status;
+  motor_operations.GetPhysicalStatus(&status);
+  EXPECT_THAT(expected, ::testing::ContainerEq(status.pos_steps));
+  motor_operations.HaltAndDiscard();
```

Comment:
To keep the readers' train of thought, I'd group the  things with newlines to better see the before/call/after situation:

```c++
  PhysicalStatus status;
  motor_operations.GetPhysicalStatus(&status);
  EXPECT_THAT(expected, ::testing::ContainerEq(status.pos_steps));

  motor_operations.HaltAndDiscard();
  EXPECT_EQ(motion_backend.clear_calls_count, 1);

  memset(expected, 0, sizeof(expected));
  motor_operations.GetPhysicalStatus(&status);
  EXPECT_THAT(expected, ::testing::ContainerEq(status.pos_steps));
```

Reply:
(solved)